helix-core/src/movement.rs

```diff
@@ -43,6 +43,47 @@ pub fn move_horizontally(
     range.put_cursor(slice, new_pos, behaviour == Movement::Extend)
 }
 
+pub fn move_horizontally_no_wrap(
+    slice: RopeSlice,
+    range: Range,
+    dir: Direction,
+    count: usize,
+    behaviour: Movement,
+) -> Range {
+    let pos = range.cursor(slice);
+
+    // Compute the new position.
+    let mut new_pos = match dir {
+        Direction::Forward => nth_next_grapheme_boundary(slice, pos, count),
+        Direction::Backward => nth_prev_grapheme_boundary(slice, pos, count),
+    };
+
+    new_pos = limit_to_line(slice, pos, new_pos, dir);
```

Comment:
I'd rather do this inside `move_horizontally` and pass in `line_wrap: bool` as a parameter. Reduces code duplication.

Reply:
The reason why I was trying to keep the same signature was because the move_impl function in commands.rs depends on the signature of move_horizontally, and then I would have to duplicate that function.

Reply:
This function is used for vertical vertical moves too and depends on the signature of `move_horizontally`, so it seems to me that you either duplicate on or the other.
https://github.com/helix-editor/helix/blob/8c29b76bccc51c19c90f7c9ee156fe2cb6f52e2c/helix-term/src/commands.rs#L485